components/driver/usb_serial_jtag.c

```diff
@@ -161,12 +162,32 @@ int usb_serial_jtag_write_bytes(const void* src, size_t size, TickType_t ticks_t
     ESP_RETURN_ON_FALSE(src != NULL, ESP_ERR_INVALID_ARG, USB_SERIAL_JTAG_TAG, "Invalid buffer pointer.");
     ESP_RETURN_ON_FALSE(p_usb_serial_jtag_obj != NULL, ESP_ERR_INVALID_ARG, USB_SERIAL_JTAG_TAG, "The driver hasn't been initialized");
 
-    const uint8_t *buff = (const uint8_t *)src;
-    // Blocking method, Sending data to ringbuffer, and handle the data in ISR.
-    xRingbufferSend(p_usb_serial_jtag_obj->tx_ring_buf, (void*) (buff), size, ticks_to_wait);
-    // Now trigger the ISR to read data from the ring buffer.
-    usb_serial_jtag_ll_ena_intr_mask(USB_SERIAL_JTAG_INTR_SERIAL_IN_EMPTY);
-    return size;
+    usb_serial_jtag_obj_t* sjtag = p_usb_serial_jtag_obj;
+
+    // try to send without blocking
+    if (xRingbufferSend(sjtag->tx_ring_buf, (void*) (src), size, 0)){
+       goto success;
```

Comment:
Nitpick: 4 spaces of indentation, and a space before the opening curly brace on the line above.

(Please try to adjust the code so it looks like the rest of the code nearby.)